### ISSUE TITLE
reverb width range discrepency between effect behavior and documentation

### ISSUE THREAD

**Issue by mrbumpy409** [FluidSynth/fluidsynth#1455]
### FluidSynth version
2.4.1

### Describe the bug
All of FluidSynth's reverb parameters have a range of 0–1, with the exception of width, which has a range of 0–100. However, the effective range for this parameter is actually 0–1, with 1 being 100% effect width. Values above 1 begin introducing an odd feedback into the reverb, and should be avoided.

At the very least, I believe this should be mentioned in the documentation, as the developers of software using FluidSynth frequently set the controls to the full 0–100 range. This often leads to control knobs where setting values between 0–1 is difficult or impossible due to the granularity of the control, and most of the usable knob range is in the weird feedback zone. QSynth has since [limited its reverb width control](https://github.com/rncbc/qsynth/issues/32) to 0–1 based on my advice, and Drumstick MIDI Player will now also be [doing the same](https://github.com/pedrolcl/drumstick/issues/26).

Here is a [test MIDI file and FluidSynth recordings](https://github.com/user-attachments/files/18222555/reverb.width.test.zip) rendered at the following reverb width settings:

* **0** – 0% stereo width in the reverb effect (mono)
* **0.5** – 50% stereo width in the reverb effect
* **1** – 100% stereo width in the reverb effect
* **5** – strange feedback/delay introduced into the effect
* **10** – even more feedback
* **50** – ...and more
* **100** – ...a crap ton of feedback

### Expected behavior
One of the following should happen, IMO:

1. Clearly document that a reverb width setting of 0–1 corresponds to 0%–100% reverb stereo width and above that goes into undesired jank territory.
2. Change the documentation to report the reverb width range as 0–1, leaving support for values up to 100 in the synthesizer for compatibility purposes.

**Comment by derselbst** [comment]
> 0 – 0% stereo width in the reverb effect (mono)
> 0.5 – 50% stereo width in the reverb effect
> 1 – 100% stereo width in the reverb effect

I don't think this is actually intended here. AFAIK, the range 0-100 exists for historical reasons, because it was the default range of the corresponding shell command. And I just applied that range when turning it into a true setting in #49. In the reverb's code there's the following comment:
https://github.com/FluidSynth/fluidsynth/blob/33f5461f83bfd6e7cf3956a28fb329ffb50039cf/src/rvoice/fluid_rev.c#L66-L69


There is also a comment about values >1:

https://github.com/FluidSynth/fluidsynth/blob/33f5461f83bfd6e7cf3956a28fb329ffb50039cf/src/rvoice/fluid_rev.c#L1043-L1045

I think whether or not a "ton of feedback" is induced also depends on the soundfont. E.g. take this tune, which I just added to the testsuite: [MIDI](https://github.com/FluidSynth/testdata/raw/refs/heads/master/reverb/DK64_sparse04.mid) + [SF2](https://github.com/derselbst/ANMP-data/raw/refs/heads/master/soundfonts/N64/DK64.sf2)  - even with width=100 it still sounds quite fine to me.

Yet I agree about the docs.  #1460 elaborates on the `synth.reverb.width` setting, pls. have a look.

---

PS: Happy new year :)

**Comment by mrbumpy409** [comment]
The update to the documentation states "A value of 100 indicates maximum separation between left and right channels.", but I would argue the maximum separation is already achieved at 1. In fact, FluidSynth's reverb width between 0 and 1 behaves exactly like other reverb engines I've used when setting between 0% and 100% effect stereo width. Going above 1 doesn't really seem to increase the stereo width, at least as I can perceive it, but instead introduces the other attributes we've already discussed (louder, almost a "doubling" type attribute to the sound).

**Comment by derselbst** [comment]
Looks like we need a third opinion. @jjceresa Happy new year to you! Since you've implemented the current FDN reverbator, what's your opinion on the meaning of the "reverb width" range?

**Comment by jjceresa** [comment]
First of all happy new year to every one. Thanks for the work done in fluidsynth !
A few points of note related to reverb width meaning.
1) The stereo unit comes directly from the previous reverb (freeverb) with width in the historical range [1..100].
outR = 0.5(1 + width) inR  + 0.5 (1 – width) inL)  
outL = 0.5(1 + width) inL  + 0.5 (1 – width) inR)

2) This unit receives at its input inR and inL which are very uncorrelated signals (i.e out of phase) resulting from FDN reverb.

>I would argue the maximum separation is already achieved at 1.

That is right, and the formula in point (1) shows that. So width sub range [0..1] meaning should be perceived as a stereo width effect with 1 being the maximum stereo width.

>Going above 1 doesn't really seem to increase the stereo width, at least as I can perceive it.

Yes, it increase the perception of uncorrelated signals inR, inL which aren't stereo channels because the reverb is a monophonic reverb. So width subrange ]1..10] meaning produces other resonnant effect.

As Tom , I noticed that  subrange [1..10] effect is input music material dependant. Sometimes it sounds fine, in other situation it sounds muddy.

It turns out that width has two meaning. As Chris said for documentation clarity perhaps subrange [0..1] should be the only one prefered for stereo width meaning.

**Comment by jjceresa** [comment]
I forgot to mention that for backward compatibility, keeping support for width up to 100 in the synthesizer would be preferable.I mean, only update the documentation should be sufficient.

**Comment by derselbst** [comment]
Thanks for clarifying JJC and confirming that this behavior is indeed intended! I have updated the linked PR accordingly, let me know if you're fine with it :)

**Comment by jjceresa** [comment]
>I have updated the linked PR accordingly, let me know if you're fine with it :)

I'am fine with this. Thanks to both of you.
